sdk/core/abort-controller/src/AbortSignal.ts

```diff
@@ -129,6 +129,18 @@ export class AbortSignal implements AbortSignalLike {
       listeners.splice(index, 1);
     }
   }
+
+  /**
+    * Dispatches a synthetic event to the AbortSignal.
+    */
+  dispatchEvent(event: Event): boolean {
```

Comment:
This is different from dispatchEvent in a possibly surprising way (not that anyone should use this API...). Might be worth either:

1. Adding a check that the event type is "abort" before invoking listeners
2. documenting that this assumes the event type is "abort"
3. just throwing that this isn't supported (IIRC the dependency is just on types)

Reply:
Thanks!  I'm fine with throwing an exception here instead of the current behavior if others think it's appropriate.  Otherwise, checking to ensure the type is `abort` before invoking also sounds like a good idea.

Reply:
I'd vote for option 1.

If you do trigger the listeners, I _think_ you'll need to trigger the `onabort` method if it is defined as well.

Reply:
Check out the improved implementation here: https://github.com/Azure/azure-sdk-for-js/pull/4756/commits/58edb7204f16a36c8937c68d51110e053b6cc58b

I've integrated approaches 1 and 3 and also moved all dispatching logic into `dispatchEvent`, adding some tests to verify it all.  Let me know if you think these `abort-controller` changes should be done in a separate PR.

Reply:
Given my other comment it might be best to port this as-is and work on fixing it up after.

Reply:
Thanks @bterlson!  I've dropped the latest commit and made the `dispatchEvent` method throw an exception.  I've also tweaked `abortSignal` to pass along a basic event object with `type: "abort"` because `core-http` is actually looking for that in its event listener.  Do you think that's OK?